src/NuGet.Clients/NuGet.CommandLine/Commands/SourcesCommand.cs

```diff
@@ -46,260 +43,42 @@ public override void ExecuteCommand()
-            }
-
-            var existingSource = SourceProvider.GetPackageSourceByName(Name);
-            if (existingSource == null)
-            {
-                throw new CommandLineException(LocalizedResourceManager.GetString("SourcesCommandNoMatchingSourcesFound"), Name);
-            }
-
-            if (!string.IsNullOrEmpty(Source) && !existingSource.Source.Equals(Source, StringComparison.OrdinalIgnoreCase))
-            {
-                if (!PathValidator.IsValidSource(Source))
-                {
-                    throw new CommandLineException(LocalizedResourceManager.GetString("SourcesCommandInvalidSource"));
-                }
-
-                // If the user is updating the source, verify we don't have a duplicate.
-                var duplicateSource = SourceProvider.GetPackageSourceBySource(Source);
-                if (duplicateSource != null)
-                {
-                    throw new CommandLineException(LocalizedResourceManager.GetString("SourcesCommandUniqueSource"));
-                }
-                existingSource = new Configuration.PackageSource(Source, existingSource.Name);
-            }
-
-            ValidateCredentials();
-
-            if (!string.IsNullOrEmpty(Username))
-            {
-                var hasExistingAuthTypes = existingSource.Credentials?.ValidAuthenticationTypes.Any() ?? false;
-                if (hasExistingAuthTypes && string.IsNullOrEmpty(ValidAuthenticationTypes))
-                {
-                    Console.WriteLine(LocalizedResourceManager.GetString("SourcesCommandClearingExistingAuthTypes"), Name);
-                }
-
-                var credentials = Configuration.PackageSourceCredential.FromUserInput(
-                    Name,
-                    Username,
-                    Password,
-                    StorePasswordInClearText,
-                    ValidAuthenticationTypes);
-                existingSource.Credentials = credentials;
-            }
-
-            SourceProvider.UpdatePackageSource(existingSource, updateCredentials: existingSource.Credentials != null, updateEnabled: false);
-
-            Console.WriteLine(LocalizedResourceManager.GetString("SourcesCommandUpdateSuccessful"), Name);
-        }
-
-        private void ValidateCredentials()
-        {
-            var isUsernameEmpty = string.IsNullOrEmpty(Username);
-            var isPasswordEmpty = string.IsNullOrEmpty(Password);
-            var isAuthTypesEmpty = string.IsNullOrEmpty(ValidAuthenticationTypes);
-
-            if (isUsernameEmpty ^ isPasswordEmpty)
-            {
-                // If only one of them is set, throw.
-                throw new CommandLineException(LocalizedResourceManager.GetString("SourcesCommandCredentialsRequired"));
-            }
-
-            if (isPasswordEmpty && !isAuthTypesEmpty)
+            SourcesAction action = SourcesAction.None;
```

Comment:
Why do we need none?

Reply:
Design guidelines for enums: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/enum

✓ DO provide a value of zero on simple enums.

Consider calling the value something like "None." If such a value is not appropriate for this particular enum, the most common default value for the enum should be assigned the underlying value of zero.

Reply:
@nkolev92 - thoughts?

Reply:
Yeah, that's ok.